backend/src/lib/repositories/user-repo.ts

```diff
@@ -34,9 +34,12 @@ class UserRepository {
       return [];
     }
 
-    return result.Users.map((cognitoUser) =>
-      UserFactory.fromCognitoUser(cognitoUser)
-    );
+    return result.Users.reduce(function (result, cognitoUser) {
+      if (cognitoUser.UserStatus !== "EXTERNAL_PROVIDER") {
+        result.push(UserFactory.fromCognitoUser(cognitoUser));
+      }
+      return result;
+    }, [] as User[]);
```

Comment:
Nit pick: Chaining `filter` + `map` might be more readable than `reduce`. Reduce is often used to return a single number (like a Sum of the elements in an array). In this case, you are trying to filter out elements that don't match a given condition, for which `filter` seems like a more natural fit. 

```js
return result.Users
  .filter(cognitoUser => cognitoUser.UserStatus !== "EXTERNAL_PROVIDER")
  .map(UserFactory.fromCognitoUser);
```

Reply:
@ferdingler Good suggestion.  I was thinking filter and map will loop through twice versus once for reduce

Reply:
@ferdingler In EnvConfig, there's now a SAML Provider field, which is the name of the IdP we configured in Cognito.  If that field is empty, then everything is as it was before the federation, which means there's no "Enterprise Sign-In" button

We don't delete the cloned Cognito user because it will be created again the next time the user logs in.  That cloned user doesn't have a role, and can't be used to login to Badger.